docs/workflows/upload-role.rst

```diff
@@ -0,0 +1,59 @@
+Upload Content
+==============
+
+Upload an Artifact to Pulp
+--------------------------
+
+This example will upload the ``geerlingguy/ansible-role-postgresql`` role. You can download that
+locally with::
+
+    curl -L https://github.com/geerlingguy/ansible-role-postgresql/archive/master.tar.gz -o pg.tar.gz
+
+
+Each Artifact in Pulp represents a file. They can be created during sync or created manually by
+uploading a file::
+
+    $ export ARTIFACT_HREF=$(http --form POST :24817/pulp/api/v3/artifacts/ file@pg.tar.gz | jq -r '._href')
+
+
+Response::
+
+    {
+        "_href": "/pulp/api/v3/artifacts/5741288b-7ed5-4199-a5a6-3d82fb97055c/",
+    }
+
+
+Reference (pulpcore): `Artifact API Usage
+<https://docs.pulpproject.org/en/3.0/nightly/restapi.html#tag/artifacts>`_
+
+
+Create a Role from the Artifact
+-------------------------------
+
+Now that Pulp has the tarball, its time to make it into a Role unit of content. We hope to add
```

Comment:
it's time

Reply:
agreed, but we need some way to determine the version. We need the tarball to contain that data somehow and that would be above the Role standard, but I think it's what is needed.